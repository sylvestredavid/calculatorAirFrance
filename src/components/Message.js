import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {Colors, IconButton} from 'react-native-paper';

export default function Message(props) {
  if (props.style === 'danger') {
    return (
      <View style={[styles.message, styles.danger]}>
        <IconButton
          icon="close"
          style={styles.close}
          size={14}
          color={Colors.white}
          onPress={props.close}
        />
        <Text style={{color: 'white'}}>{props.text}</Text>
      </View>
    );
  } else {
    return (
      <View style={[styles.message, styles.success]}>
        <IconButton
          icon="close"
          style={styles.close}
          size={14}
          color={Colors.white}
          onPress={props.close}
        />
        <Text style={{color: 'white'}}>{props.text}</Text>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  message: {
    padding: 15,
    marginTop: 5,
    marginBottom: 20,
    borderRadius: 5,
    alignSelf: 'flex-start',
    position: 'relative',
  },
  danger: {
    backgroundColor: 'red',
  },
  success: {
    backgroundColor: 'green',
  },
  close: {
    position: 'absolute',
    top: -5,
    right: -5,
  },
});
