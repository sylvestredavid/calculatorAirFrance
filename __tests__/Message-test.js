/**
 * @format
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Message from '../src/components/Message';

test('renders correctly', () => {
  const tree = renderer.create(<Message />).toJSON();
  expect(tree).toMatchSnapshot();
});
