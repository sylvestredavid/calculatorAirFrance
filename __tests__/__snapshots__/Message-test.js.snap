// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderRadius": 5,
        "marginBottom": 20,
        "marginTop": 5,
        "padding": 15,
        "position": "relative",
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={false}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 10.5,
            "height": 21,
            "width": 21,
          },
          undefined,
          Object {
            "position": "absolute",
            "right": -5,
            "top": -5,
          },
        ],
      ]
    }
  >
    <View>
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            Array [
              Object {
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°…–
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  />
</View>
`;
